src/graphql.js

```diff
@@ -39,14 +39,19 @@ import type { GraphQLSchema } from './type/schema';
  *    The name of the operation to use if requestString contains multiple
  *    possible operations. Can be omitted if requestString contains only
  *    one operation.
+ * logFn:
+ *    The function that is called with a tag of an event, an event payload
+ *    and other execution information.
+ *    The examples include: errors thrown, before and after the resolver call.
  */
 export function graphql(
   schema: GraphQLSchema,
   requestString: string,
   rootValue?: mixed,
   contextValue?: mixed,
   variableValues?: ?{[key: string]: mixed},
-  operationName?: ?string
+  operationName?: ?string,
+  logFn?: (tag: string, payload: mixed, info: mixed) => void
```

Comment:
It seems like there are two main cases that this PR will need to account for:

- Timing information
- Inspection of poorly performing resolvers

_**Timings**_
Since the logging of `*_START` and `*_END` are separate, in order to measure resolver timings similar to what Apollo Optics does, this function should probably guarantee either a key like `query:<path>:<fieldName>` or enough information about a resolver to construct one. I see that it accepts path (in some places) already, but that is likely not quite enough, especially if multiple queries are running at once.

_**Inspection**_
My guess is that it will eventually be useful to folks to log all sorts of information about poorly performing resolvers, much more than just the path or key. For example, if an Array resolver has a search/filter argument, it would be helpful to know which filters are running slowly. Rather than a payload and info of type `mixed`, why not just pass the same or similar objects to what a resolver function receives?